Repository: plone.app.upgrade


Branch: refs/heads/master
Date: 2022-06-24T19:36:37+02:00
Author: Maurits van Rees (mauritsvanrees) <maurits@vanrees.org>
Commit: https://github.com/plone/plone.app.upgrade/commit/7cc4d2a3ef9b62640842779a53224014594eb8d8

update_catalog_metadata: catch and log ComponentLookupError when getting indexable object.

Files changed:
A news/3521.bugfix
M plone/app/upgrade/utils.py

b'diff --git a/news/3521.bugfix b/news/3521.bugfix\nnew file mode 100644\nindex 00000000..1fd110da\n--- /dev/null\n+++ b/news/3521.bugfix\n@@ -0,0 +1,2 @@\n+``update_catalog_metadata``: catch and log ``ComponentLookupError`` when getting indexable object.\n+[maurits]\ndiff --git a/plone/app/upgrade/utils.py b/plone/app/upgrade/utils.py\nindex 6a7bc0c5..2541eec8 100644\n--- a/plone/app/upgrade/utils.py\n+++ b/plone/app/upgrade/utils.py\n@@ -12,6 +12,7 @@\n from Products.ZCatalog.ProgressHandler import ZLogHandler\n from types import ModuleType\n from ZODB.POSException import ConflictError\n+from zope.component import ComponentLookupError\n from zope.component import getMultiAdapter\n \n import logging\n@@ -389,7 +390,12 @@ def update_catalog_metadata(context, column=None):\n             record = metadata[rid]\n             old_value = record[column_position]\n             # see CMFPlone/catalog.zcml\n-            wrapper = getMultiAdapter((obj, catalog), IIndexableObject)\n+            try:\n+                wrapper = getMultiAdapter((obj, catalog), IIndexableObject)\n+            except ComponentLookupError:\n+                # I have seen the portal_catalog as brain in the portal_catalog...\n+                logger.exception(obj)\n+                continue\n             # See ZCatalog/Catalog.py:recordify\n             new_value = getattr(wrapper, column, MV)\n             if (new_value is not MV and safe_callable(new_value)):\n'

Repository: plone.app.upgrade


Branch: refs/heads/master
Date: 2022-06-24T19:38:27+02:00
Author: Maurits van Rees (mauritsvanrees) <m.van.rees@zestsoftware.nl>
Commit: https://github.com/plone/plone.app.upgrade/commit/b518410043f7f6b5c4ed62d92000ff4541e048e8

Merge pull request #293 from plone/maurits-componentlookuperror

update_catalog_metadata: catch ComponentLookupError

Files changed:
A news/3521.bugfix
M plone/app/upgrade/utils.py

b'diff --git a/news/3521.bugfix b/news/3521.bugfix\nnew file mode 100644\nindex 00000000..1fd110da\n--- /dev/null\n+++ b/news/3521.bugfix\n@@ -0,0 +1,2 @@\n+``update_catalog_metadata``: catch and log ``ComponentLookupError`` when getting indexable object.\n+[maurits]\ndiff --git a/plone/app/upgrade/utils.py b/plone/app/upgrade/utils.py\nindex 6a7bc0c5..2541eec8 100644\n--- a/plone/app/upgrade/utils.py\n+++ b/plone/app/upgrade/utils.py\n@@ -12,6 +12,7 @@\n from Products.ZCatalog.ProgressHandler import ZLogHandler\n from types import ModuleType\n from ZODB.POSException import ConflictError\n+from zope.component import ComponentLookupError\n from zope.component import getMultiAdapter\n \n import logging\n@@ -389,7 +390,12 @@ def update_catalog_metadata(context, column=None):\n             record = metadata[rid]\n             old_value = record[column_position]\n             # see CMFPlone/catalog.zcml\n-            wrapper = getMultiAdapter((obj, catalog), IIndexableObject)\n+            try:\n+                wrapper = getMultiAdapter((obj, catalog), IIndexableObject)\n+            except ComponentLookupError:\n+                # I have seen the portal_catalog as brain in the portal_catalog...\n+                logger.exception(obj)\n+                continue\n             # See ZCatalog/Catalog.py:recordify\n             new_value = getattr(wrapper, column, MV)\n             if (new_value is not MV and safe_callable(new_value)):\n'

